backend/utils/fileUpload.ts

```diff
@@ -0,0 +1,113 @@
+import { Request } from 'express';
+import multer from 'multer';
+import path from 'path';
+
+// Configure the file storage
+const storage = multer.diskStorage({
```

Comment:
I don't actually know the best way of doing this, this is just what I've done in the past with cloudinary. I agree that if we can avoid uploading to the server that would be preferable. I can look into it

Reply:
Ok yeah we can use multer.memoryStorage() instead of diskStorage. It will store the file in memory as a buffer and then send the buffer to S3.
Something similar to this: https://github.com/meech-ward/s3-get-put-and-delete/blob/master/express-react/express/server.js

Reply:
Another option is using pre-signed urls and uploading directly from the client to S3